code/game/objects/items/devices/illegalradio.dm

```diff
@@ -48,34 +52,43 @@
 	user.set_machine(src)
 	interact(user)
 
+	
+
 /obj/item/device/illegalradio/Topic(href, href_list)
 	..()
 
 	if (href_list["buy_item"])
-		var/item = href_list["buy_item"]
-		var/list/split = splittext(item, ":") // throw away variable
+		var/delivery_method = href_list["buy_item"]
+		delivery_method = text2num(delivery_method)
+		
+		if(!selected_item)
+			return 0
+
+		selected_item.buy(src, delivery_method, usr)
+		selected_item = null
+		interact(usr)
+				
+	else if(href_list["open_delivery_menu"])
+		var/input = href_list["open_delivery_menu"]
+		var/list/split = splittext(input, ":")
 
 		if(split.len == 2)
-			// Collect category and number
 			var/category = split[1]
 			var/number = text2num(split[2])
 
 			var/list/buyable_items = get_black_market_items()
-
-			var/list/black_market = buyable_items[category]
-			if(black_market && black_market.len >= number)
-				var/datum/black_market_item/I = black_market[number]
-				if(I)
-					I.buy(src, usr)
-					if(!nanotrasen_variant && prob(I.sps_chance))
-						SPS_black_market_alert(src, "The SPS decryption complex has detected an illegal black market purchase of item [I.name]")
-			else
-				var/text = "[key_name(usr)] tried to purchase a black market item that doesn't exist."
-				var/textalt = "[key_name(usr)] tried to purchase a black market item that doesn't exist: [item]."
-				message_admins(text)
-				log_game(textalt)
-				log_admin(textalt)
-
+			var/list/category_items = buyable_items[category]
+			
+			if(category_items && category_items.len >= number)
+				var/datum/black_market_item/item = category_items[number]
+				if(item)
```

Comment:
again, add feedback if this fails

Reply:
This isn't a concern since it'll just be an empty section (which tells you its empty inof itself), but realistically it'll never even happen anyway.